phpBB/phpbb/content_visibility.php

```diff
@@ -144,7 +144,13 @@ public function get_count($mode, $data, $forum_id)
 	*/
 	public function is_visible($mode, $forum_id, $data)
 	{
-		$is_visible = $this->auth->acl_get('m_approve', $forum_id) || $data[$mode . '_visibility'] == ITEM_APPROVED;
+		$visibility = $data[$mode . '_visibility'];
+		$poster_key = ($mode === 'topic') ? 'topic_poster' : 'poster_id';
+		$is_visible = $this->auth->acl_get('m_approve', $forum_id) || $visibility == ITEM_APPROVED;
+		$is_visible = $is_visible || (
+			($visibility == ITEM_UNAPPROVED || $visibility == ITEM_REAPPROVE)
+			&& $this->user->data['user_id'] === $data[$poster_key]
```

Comment:
if I'm not mistaken, this is gonna show unapproved anonymous posts to anonymous users, right?

Reply:
As it is now yes. I added it to the TODO list to handle it. I'm really just trying to mock up what changes are needed first so I can have a better idea on what changes would the database tables need to make this a bit more optimized (as of now all users would have a separate query which doesn't seem optimal).

Reply:
``$this->user->data['user_id'] === $data[$poster_key]`` first? If is not the user, it not performs the 2 checks.